setup.py

```diff
@@ -1,6 +1,8 @@
 import os
+import shutil
 
 from setuptools import setup
 
-os.system('g++ -std=c++11 -O3 scib/knn_graph/knn_graph.cpp -o scib/knn_graph/knn_graph.o')
+if shutil.which('g++') is not None:
+    os.system('g++ -std=c++11 -O3 scib/knn_graph/knn_graph.cpp -o scib/knn_graph/knn_graph.o')
```

Comment:
could you add an error report here as well?

Reply:
Agreed.